### ISSUE TITLE
Auto-Launch not working when run from npm

### ISSUE THREAD

**Issue by BrianGenisio** [testem/testem#654]
I just updated to Node 5.0.0 with NPM 3.3.6.  I'm running on OSX 10.10.5 (Yosemite)

I have a "test" script defined in my project.json:

``` json
"scripts": {
    "test": "testem",
    "test-ci": "testem ci --port 17357"
  }
```

So when I run `npm test`, it starts testem.  

The problem is that my `launch_in_dev` browser (Chrome) is not auto-run.  Even if I call it out explicitly: `"test": "testem -l Chrome"`, the browser does not auto-launch.  Testem DOES start, but the browser does not.  I can run it by hand, and that works.  It's just odd.

Running the tests by hand: `./node_modules/.bin/testem` DOES work... it just seems like this use case is being blocked.  And, this used to work (node 0.12 + NPM 2).

Does anyone know a workaround?  I try to avoid installing testem globally and prefer to do everything via `npm run` commands.

**Comment by BrianGenisio** [comment]
I have a workaround... it is a hack, but it works for now...

```json
"test": "open http://localhost:7357 && testem"
```

**Comment by johanneswuerbach** [comment]
Maybe npm 3 is somehow fiddling with the `$PATH`, but this definitely needs further investigation.

**Comment by BrianGenisio** [comment]
Yeah, I suspect this is an NPM thing, and less of a testem problem, but running testem from npm scripts is a very common workflow, so it is good to get a handle on it.  

Also, some more data points: PhantomJS does launch properly when I run `npm test` configured that way.

**Comment by revolunet** [comment]
`testem ci` keeps waiting and doesnt display an error

with `--debug` flag i can see this error, and the provided Chrome path does not exist :
```
WARN Chrome errored   errno: 'ENOENT',
WARN Chrome errored   syscall: 'spawn /Users/juju/Applications/Google Chrome.app/Contents/MacOS/Google Chrome',
```

**Comment by revolunet** [comment]
(works with firefox though)

**Comment by revolunet** [comment]
ok, once  Chrome exists in one of these locations, it works as expected (osx)

**Comment by rupurt** [comment]
@BrianGenisio @revolunet I'm encountering the same issue :frowning:. CI hangs and unfortunately opening the browser before running testem doesn't work because the server isn't running by that stage.

**Comment by revolunet** [comment]
I changed my chrome path to the "standard" and it solved the issue; looks like this is hardcoded here : https://github.com/airportyh/testem/blob/f55c8e77c1643ba15f3288e1158eaff81a23e19b/lib/browser_launcher.js#L136


````
scripts: {
    "test": "./node_modules/.bin/testem ci",
    "test-chrome": "./node_modules/.bin/testem -l chrome",
},
devDependencies: {
    "react-addons-test-utils": "^0.14.1",
    "testem": "^0.9.8",
    ...
}
```

**Comment by BrianGenisio** [comment]
FWIW, I upgraded to El Capitan this weekend, and the auto-launcher magically started working again.

**Comment by rupurt** [comment]
@revolunet can you elaborate on `I changed my chrome path to the "standard"`?

**Comment by revolunet** [comment]
yes, i had Chrome in a custom path but changed it to `/Applications/Google Chrome.app/Contents/MacOS/Google Chrome` as defined in testem runner source and it worked

**Comment by rupurt** [comment]
@revolunet unfortunately mine is already install at the standard location. @BrianGenisio maybe it's time to upgrade to El Capitan :grimacing:

**Comment by rupurt** [comment]
@BrianGenisio @revolunet I figured out my problem :smile: :smile: :smile: 

I'm using webpack to compile my assets and was using the watch flag. That meant that the `before_tests` hook never finished and never ran any of the launchers.

**Comment by johanneswuerbach** [comment]
Thanks @rupurt for that feedback, the UX around which hooks are currently running definitely needs improvement.

**Comment by rupurt** [comment]
@johanneswuerbach no worries. Would you like a PR updating the README with Webpack setup instructions?

**Comment by johanneswuerbach** [comment]
I think https://github.com/airportyh/testem/tree/master/examples would be a better place, but definitely would welcome that.

**Comment by rupurt** [comment]
:+1: 

@johanneswuerbach do you have any recommendations on how I would get Webpack to recompile on file changes now that I have to run the hook without watching?

**Comment by johanneswuerbach** [comment]
Sorry, but I never worked with webpack.

**Comment by rupurt** [comment]
Do you know how Ember-CLI & Browserify do it? It's not clear to me from first glance at the examples.

**Comment by johanneswuerbach** [comment]
ember-cli has its own watcher and triggers testem programatically on change https://github.com/ember-cli/ember-cli/blob/3c615aa55dfe6be1adf14ef67e74028b09560138/lib/tasks/test-server.js#L41

**Comment by rupurt** [comment]
Ok cheers for the lead. It would be cool if there was a hook that could be triggered when any of the `src_files` changed. That's kind of what I expected `before_tests` to do.

How hard do you think that would be to add? If it's reasonable I could take a stab at it.

**Comment by et** [comment]
Sorry to revive a dead thread, but did anyone find a solution for this?
I tried to integrate with sauce labs and Bamboo and noticed that the tests failing to start. Then I started a manual test into the tunnel and visited http://localhost:7357 and they started running.

**Comment by thomasmarshall** [comment]
I managed to track this down to an older version of `node-which` that seems to mistakenly match any path starting with `/`.

`testem` looks in two possible locations for Chrome, and if `which` finds the first one, then `testem` uses that. In my case the faulty version of `node-which` responds that `/Users/tom/Applications/Google Chrome.app/Contents/MacOS/Google Chrome` exists, even though it doesn't.

I fixed by installing a newer version: `npm install which`
